code/lightna/magento-os-frontend/js/common/ShoppingCart.js

```diff
@@ -1,121 +1,89 @@
-                }
-            },
-            showMessage: false,
+        $$('[data-action="remove-product"]').foreach((i, trigger) => {
+            const itemId = trigger.getAttribute('data-item-id');
+            trigger.addEventListener('click', () => {
+                this.removeProduct(itemId);
+            });
         });
     }
 
-    removeProduct(itemId) {
-        const data = {
-            item_id: itemId,
-        };
-        const itemToRemove = $(`[data-item-id="${itemId}"]`, $('.cjs-minicart-content')).closest('li');
-
-        request.post('/checkout/sidebar/removeItem/', data, {
-            onSuccess: () => {
-                itemToRemove.classList.add('fade-out');
-                setTimeout(() => {
-                    this.updateMinicart.bind(this)();
-                }, 200);
+    update() {
+        request.post(
+            this.shoppingCartBlockUrl,
+            {},
+            {
+                onSuccess: this.redraw.bind(this),
             },
+        ).then(() => {
+            this.open();
         });
     }
 
-    updateMinicart() {
-        const updateMinicartUrl = document.location.pathname + '?block=.body.minicart-content';
-
-        request
-            .post(
-                updateMinicartUrl,
-                {},
-                {
-                    onSuccess: this.updateMinicartHtml.bind(this),
-                },
-            )
-            .then(() => {
-                this.updateMagentoCartSectionId();
-            });
-    }
-
-    updateMinicartHtml(html) {
-        $('.cjs-minicart-content').outerHTML = html;
-        this.addCartEventListeners();
-        this.updateCounter();
-    }
-
-    updateMagentoCartSectionId() {
-        let sids = Cookie.get('section_data_ids');
-        sids = sids ? JSON.parse(decodeURIComponent(sids)) : {};
-        sids.cart = sids.cart ? sids.cart + 1000 : 1;
-
-        Cookie.set('section_data_ids', encodeURIComponent(JSON.stringify(sids)));
+    redraw(html) {
+        $(this.shoppingCart).outerHTML = html;
+        this.bindCartActionsEvents();
     }
 
-    updateCounter() {
-        $('.cjs-minicart-counter').innerText = $('.minicart').dataset.totalQty;
+    open() {
+        setTimeout(() => {
+            document.body.classList.add(this.classes.cartOpen);
+        }, this.minActionDuration);
     }
 
-    openCart() {
-        document.body.classList.add('minicart-open');
+    close() {
+        document.body.classList.remove(this.classes.cartOpen);
     }
 
-    closeCart() {
-        document.body.classList.remove('minicart-open');
+    removeProduct(itemId) {
+        const data = {
+            item_id: itemId,
+        };
+        const itemToRemove = $(
+          `[data-item-id="${itemId}"]`,
+          $(this.shoppingCart),
+        ).closest('li');
+
+        request.post(this.removeFromCartUrl, data, {
+            onSuccess: this.removeProductSuccessHandler.bind(this, itemToRemove),
```

Comment:
removeProductSuccessHandler - confises, sound like function removes a success handler.
Ideas:
* handleRemoveProductSuccess
* onProductRemove
 
hmm, second is short